### PR TITLE
Make repo compliant

### PR DESCRIPTION
**Pull request by RaphaelVogel**
**What this PR does / why we need it**:
* Deleted `NOTICE.md` file
* Updated `CONTRIBUTING.md` file
* Created/Updated `LICENSE` file
* Created/Updated `.reuse/dep5` file
* Created/Updated `LICENSES` folder
* Added badge for reuse compliance in `README.md` file and registered repo with REUSE API
* License/Copyright header replacement for 'go' files
  * hack/tools.go

### CODE REVIEW

**Comment by RaphaelVogel** [comment]
@ScheererJ : It's a pleasure for me to make this PR for you to be reviewed

File: .reuse/dep5 [hunk 1]
```diff
@@ -2,65 +2,30 @@ Format: https://www.debian.org/doc/packaging-manuals/copyright-format/1.0/
 Upstream-Name: Gardener ext-authz-server
 Upstream-Contact: The Gardener project <gardener@googlegroups.com>
 Source: https://github.com/gardener/ext-authz-server
+
```
Comment (ScheererJ) [inline .reuse/dep5:5]: Why did you add an empty, new line? Are you paid by lines of code?

File: .reuse/dep5 [hunk 2]
```diff
@@ -2,65 +2,30 @@ Format: https://www.debian.org/doc/packaging-manuals/copyright-format/1.0/
...
 License: Apache-2.0
 
 # --------------------------------------------------
 # documentation
 
-Files:
-  *.md
-Copyright: 2021 SAP SE or an SAP affiliate company and Gardener contributors
+Files: *.md
+Copyright: 2017-2024 SAP SE or an SAP affiliate company and Gardener contributors
```
Comment (ScheererJ) [inline .reuse/dep5:17]: Why do we use a different license for our markdown files compared with the source code?

**Review by ScheererJ — COMMENTED** [review]
Closing a pull request with open questions does not make the questions go away. Please answer/reply to the questions/comment.

File: .reuse/dep5 (hunk 1 above)
Comment (RaphaelVogel) [reply .reuse/dep5:5]: it just looks better

File: .reuse/dep5 (hunk 2 above)
Comment (RaphaelVogel) [reply .reuse/dep5:17]: It's recommended by OSPO guidelines

File: .reuse/dep5 (hunk 1 above)
Comment (ScheererJ) [reply .reuse/dep5:5]: We are not at a beauty pageant. Therefore, I ask again: What is the technical benefit of having this empty line?

Comment (RaphaelVogel) [reply .reuse/dep5:5]: Let ChatGPT answer this:

Blank lines in programming are used for several purposes:
1. **Readability**: Blank lines help to break up the code into logical sections, making it easier to read and understand. They provide visual cues about the structure of the code.
2. **Organization**: They can be used to separate different parts of the code such as classes, methods, loops, conditionals, etc. This helps in organizing the code better.
3. **Style Guidelines**: Many programming languages have style guidelines that recommend the use of blank lines. For example, in Python, PEP 8 style guide recommends using blank lines in certain situations to indicate logical sections of the code.
4. **Documentation**: In some languages like Python, blank lines can be used in conjunction with comments to provide documentation for the code.
5. **Separation of Import Statements**: In many languages, blank lines are used to separate import statements from the rest of the code.
Remember, the use of blank lines can vary depending on the programming language and the specific style guidelines being followed. However, the overarching goal is to improve the readability and maintainability of the code.

**Comment by ghost** [comment]
@axel7born, @docktofuture You have pull request review open invite, please check

File: .reuse/dep5 (hunk 1 above)
Comment (ScheererJ) [reply .reuse/dep5:5]: @RaphaelVogel Thanks a lot for these insights. Could you now please comment why YOU added the empty line? I hope this line was not generated by ChatGPT as that might have unfortunate copyright implications.

Comment (RaphaelVogel) [reply .reuse/dep5:5]: Readability: To separate header information in `dep5` from the actual data ;-)

**Review by ScheererJ — APPROVED** [review]
/lgtm

**Pull request merged**